test/csv/csv-filter/test_csv_filter.rb

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+# frozen_string_literal: false
+
+require_relative '../helper'
+
+require 'csv'
+
+class TestFilter < Test::Unit::TestCase
+
+  # Some rows data (useful as default).
+  Rows = [
+    %w[aaa bbb ccc],
+    %w[ddd eee fff],
+  ]
+
+  def setup
+    # In case the previous test left this as true.
+    $TEST_DEBUG = false
+  end
+
+  # Print debugging information if indicated.
+  def debug(label, value, newline: false)
+    return unless $TEST_DEBUG
+    print("\n") if newline
+    printf("%15s: %s\n", label, value.inspect)
+  end
+
+  # Return the test name (retrieved from the call stack).
+  def get_test_name
+    caller.each do |x|
+      method_name = x.split(' ').last.gsub(/\W/, '')
+      return method_name if method_name.start_with?('test')
+    end
+    raise RuntimeError.new('No test method name found.')
+  end
+
+  # Perform the testing defined in the caller's block.
+  def do_test(debugging: false)
+    # Just the caller's block, unless debugging.
+    unless debugging
+      yield
+      return
+    end
+    # Wrap the caller's block with debugging information.
+    $TEST_DEBUG = true
+    test_name = get_test_name
+    debug('BEGIN', test_name, newline: true)
+    yield
+    debug('END', test_name)
+    $TEST_DEBUG = false
+  end
+
+  # Return CSV string generated from rows array and options.
+  def make_csv_s(rows: Rows, **options)
+    csv_s = CSV.generate(**options) do|csv|
+      rows.each do |row|
+        csv << row
+      end
+    end
+    csv_s
+  end
+
+  # Return filepath of file containing CSV data.
+  def csv_filepath(csv_in_s, dirpath, option_sym)
+    filename = "#{option_sym}.csv"
+    filepath = File.join(dirpath, filename)
+    File.write(filepath, csv_in_s)
+    filepath
+  end
+
+  # Return stdout and stderr from CLI execution.
+  def execute_in_cli(filepath, cli_options_s = '')
```

Comment:
* How about renaming this to `run_csv_filter` or something?
* Why do we need to use string not array for options?

Reply:
Renamed; modified to use array.